language/stdlib/transaction_scripts/register_validator.move

```diff
@@ -17,5 +23,11 @@ fun main(
       validator_network_address,
       fullnodes_network_identity_pubkey,
       fullnodes_network_address
-  )
+  );
+
+  let sender = Transaction::sender();
+  // Validating nodes need to accep all currencies in order to receive txn fees
```

Comment:
```suggestion
  // Validating nodes need to accept all currencies in order to receive txn fees
```

Reply:
Wow, good catch! 

- We might want to move this logic into `ValidatorConfig::register_candidate_validator` or the candidate validator approval code (if the dependencies work out) to make it extra explicit that this is a precondition for being  a validator
- We need to think carefully about what the upgrade path for adding a new currency to Libra looks like for existing validators--they had better accept it before we mint any!

Reply:
1. Agreed, I kinda went back-and-forth on it. But I'm not entirely sure if we can fully say accepting all currencies is a precondition for a validator (it could be that if they don't accept the currency then we don't send any txn fees in that currency to them, and it gets distributed amongst the other validators). 
2. Totally agreed about what it looks like adding a new currency to the network. 

If you're OK with it, I'd like to hold off answering these questions for this PR since in general I think there's an important story around validators and txn fees that needs to be examined, and that I'm planning on digging into shortly. As an example of another interesting question related to this area is what account type should validators be in mainnet? I presume they shouldn't be a VASP-related account, they can't be `Empty` because they need to hold the collected fees, and they can't be `Unhosted` due to txn limits.